### PR TITLE
Introduced MC truth information for EMCal

### PR DESCRIPTION
**Pull request by aknospe**
Introduced EMCal MCLabel class, modified Digit and Digitizer classes to work with this class to store MC truth information, changed tower index in Digit to Short_t.

### CODE REVIEW

File: Detectors/EMCAL/base/include/EMCALBase/Digit.h [hunk 1]
```diff
@@ -51,19 +52,23 @@ class Digit : public DigitBase
...
+  Int_t GetLabel() const { return mLabel; }
+  void SetLabel(Int_t label) { mLabel = label; }
 
   void PrintStream(std::ostream& stream) const;
 
  private:
   friend class boost::serialization::access;
 
-  Int_t mTower;        ///< Tower index (absolute cell ID)
+  Short_t mTower;      ///< Tower index (absolute cell ID)
```
Comment (sawenzel) [inline Detectors/EMCAL/base/include/EMCALBase/Digit.h]: please arrange data member in decreasing size/alignment order. This should make the Digit smaller.

File: Detectors/EMCAL/base/include/EMCALBase/Digit.h [hunk 2]
```diff
@@ -51,19 +52,23 @@ class Digit : public DigitBase
...
   Double_t mAmplitude; ///< Amplitude
+  Int_t mLabel;        ///< Index of the corresponding entry/entries in the MC label array
```
Comment (sawenzel) [inline Detectors/EMCAL/base/include/EMCALBase/Digit.h]: is this cross-reference needed?

File: Detectors/EMCAL/simulation/include/EMCALSimulation/MCLabel.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#ifndef ALICEO2_EMCAL_MCLABEL_H_
+#define ALICEO2_EMCAL_MCLABEL_H_
+
+#include "SimulationDataFormat/MCCompLabel.h"
+
+namespace o2
+{
+namespace EMCAL
+{
+class MCLabel : public o2::MCCompLabel
```
Comment (sawenzel) [inline Detectors/EMCAL/simulation/include/EMCALSimulation/MCLabel.h]: so do I understand correctly that this label is only used temporarily and serves to backup the event time of the eventID contributing to a digit?

File: Detectors/EMCAL/base/include/EMCALBase/Digit.h (hunk 1 above)
Comment (aknospe) [reply Detectors/EMCAL/base/include/EMCALBase/Digit.h]: OK, I can fix it.

File: Detectors/EMCAL/base/include/EMCALBase/Digit.h (hunk 2 above)
Comment (aknospe) [reply Detectors/EMCAL/base/include/EMCALBase/Digit.h]: Yes, I think so. We need some way to connect the Digit to the MCTruth record of the hits that contributed to it. I used the TOF code as a guide and this is how they did it.

File: Detectors/EMCAL/simulation/include/EMCALSimulation/MCLabel.h (hunk 3 above)
Comment (aknospe) [reply Detectors/EMCAL/simulation/include/EMCALSimulation/MCLabel.h]: The labels associated with each Digit are sorted according to the event time, which is why we are storing that quantity in this class. Again, I based this on the TOF code, which sorts based on the TDC value. @mfasDa said that we want to be sorting on the event time.

**Comment by sawenzel** [comment]
Can you please also fix the formatting?

**Comment by aknospe** [comment]
I have reordered the data members in the Digit class. If there are further formatting issues, please let me know.

**Comment by sawenzel** [comment]
clang-format is complaing. See here: https://travis-ci.org/AliceO2Group/AliceO2/jobs/469578164

**Comment by aknospe** [comment]
OK, I took care of the clang-format complaints.

**Comment by sawenzel** [comment]
Thanks. What remains to be done is an integration into the DPL workflow (passing forward labels to the output device).

**Pull request merged**